app/masterhook.c

```diff
@@ -40,9 +40,9 @@ int drmModeSetCrtc(int fd, uint32_t crtcId, uint32_t bufferId,
                    uint32_t x, uint32_t y, uint32_t *connectors, int count,
                    drmModeModeInfoPtr mode)
 {
-    // Grab the first DRM FD that makes it in here. This will be the Qt
-    // EGLFS backend's DRM FD, which we will need later.
-    if (g_QtDrmMasterFd == -1) {
+    // Grab the first DRM Master FD that makes it in here. This will be the Qt
+    // EGLFS backend's DRM FD, on which we will call drmDropMaster() later.
+    if (g_QtDrmMasterFd == -1 && drmIsMaster(fd)) {
```

Comment:
We should probably do this for the atomic case below too.

Reply:
Good idea -- done.